communications/events/messages.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from dataclasses import dataclass
+
+from asgiref.sync import async_to_sync
+from channels.db import database_sync_to_async
+from channels.layers import get_channel_layer
+
+from communications.entities.messages import Message
+from communications.events.base import BaseEvent
+
+
+logger = logging.getLogger(__name__)
+
+@dataclass
+class MessageNotificationEvent(BaseEvent):
+    async def trigger(self, message: Message):
+        from notifications.models import Notification  # Import delayed
+        try:
+            await database_sync_to_async(Notification.objects.create)(
+                investor_id=message.sender_id,
+                startup_id=message.receiver_id,
+            )
+
```

Comment:
Currently, all errors in the trigger method are logged as "Failed to create and send notification," which doesn’t specify whether the error occurred during the database operation or during the group_send.

try:
    await database_sync_to_async(Notification.objects.create)(
        investor_id=message.sender_id,
        startup_id=message.receiver_id,
    )
    logger.info(f"Notification object created for message {message.content}")
except Exception as e:
    logger.error(f"Failed to create Notification: {e}", exc_info=True)
    return  # Prevent further steps if creation fails

Reply:
Done